V3-Upgrade-Guide.md

```diff
@@ -221,11 +221,47 @@ The `serviceMapping` option in OData generator has been renamed to `optionsPerSe
 The `service-mapping.json` file is now `options-per-service.json`.
 The file will not be generated by default in the input directory, as in the previous versions.
 
-Set the `optionsPerService` to either:
+You **must** set  the `optionsPerService` to have the generator consider the options. 
+You can set the value to either:
 
 - The file path containing the options per service (e.g. `options.json`).
-  If the file does not exist, it will be created.
-- The directory from which the file is read/created (e.g. `someDir`. This will create a file named `options-per-service.json` in `someDir`)
+  If the file does not exist, it will be created and initialized.
+  If the file exists, missing or partial service options will be added with the default values. 
+- The directory from which the file is read/created (e.g. `someDir`. This will read/create a file named `options-per-service.json` in `someDir`)
```

Comment:
how does the generator distinguish directories and files? do files have to end in `.json`? Do directories have to have a trailing `/`?

Reply:
If a dirPath is given it will read/create a file `options-per-service.json` in that dir. If a filePath is given it read/create a file with a given name.